### DISCUSSION TITLE
Temporary Library Question

### DISCUSSION THREAD

**Discussion by reividelacruz** [codefori/vscode-ibmi#263] [Q&A] [answered]
If creating new libraries on an IBM i system is restricted and only one library is assigned to a developer, can that library also be the "Temporary Library"? Will existing objects present there be removed by the extension at a particular point in time?

**Comment by worksofliam** [comment, marked as answer]
> can that library also be the "Temporary Library"?

Absolutely! Any library can be the temporary one.

> Will existing objects present there be removed by the extension at a particular point in time?

No. Files are not deleted automatically. We suggest to uses that have a shared temp library automatically run `DLTOBJ TMPLIB/*ALL *FILE` regularly since all the extension does is create temp files.

**Reply by reividelacruz** [reply]
Great! Thank you!
